snowflake/datadog_checks/snowflake/check.py

```diff
@@ -53,7 +53,9 @@ def __init__(self, *args, **kwargs):
             self.register_secret(self._config.password)
 
         if self._config.private_key_password:
-            self.register_secret(self._config.private_key_password)
+            # private_key_password is of type bytes so it can be loaded for the pem private key
+            # but this needs to converted to string to be registered as a secret for filtering
+            self.register_secret(str(self._config.private_key_password))
```

Comment:
Nice! Nit- should we update the tests so the private key password is not a string?

Reply:
We automatically change it to `bytes` in the config: https://github.com/DataDog/integrations-core/blob/e29813bb0a29b945c57e897e3529d7cf26524c59/snowflake/datadog_checks/snowflake/config.py#L45

Reply:
Should do `ensure_unicode` https://github.com/DataDog/integrations-core/blob/e29813bb0a29b945c57e897e3529d7cf26524c59/datadog_checks_base/datadog_checks/base/__init__.py#L9

Reply:
Ah I see now! Do you know why the tests didn't catch this?

Reply:
I don't think our tests caught this because `private_key_password` isn't as a config option in any happy path dd_run_check() tests. I can add a test that uses private key after merging this PR though.

Reply:
Is it quick enough to add now? Also do you think it would make more sense to keep the password as is, then only convert it to bytes when using to avoid extra conversion?